Graphics/ShaderTools/include/ShaderToolsCommon.hpp

```diff
@@ -80,15 +80,24 @@ const char* ReadShaderSourceFile(const char*                      SourceCode,
 /// Appends shader source code to the source string
 void AppendShaderSourceCode(std::string& Source, const ShaderCreateInfo& ShaderCI) noexcept(false);
 
-
-struct ProcessShaderIncludesInfo
+struct IncludePreprocessInfo
 {
-    IDataBlob*  DataBlob = nullptr;
-    const char* FilePath = nullptr;
+    /// The function to let the application to get data of the included file
+    std::function<void(IDataBlob* pDataBlob, const Char* FilePath)> IncludeHandler{};
+
+    /// The function to let the application to modify #include "${PATH}" substring in the source file
+    ///
+    ///      #include "SomeInclude.hlsl"
+    ///      ^                         ^
+    ///    Start                      End
+    std::function<void(Char* Start, Char* End)> IncludeModify{};
 };
```

Comment:
I think it is better to add `Char* Start; Char* End;` to `ProcessShaderIncludesInfo` struct so that there is only one callback function that can do what it needs with data and modify the string at the same time.

It is also not clear if IncludeHandler returns data before modification or after - this will be clear if it was one function.

Also I think that End is the position *past* the last symbol, not the last symbol itself

```
    ///      #include "SomeInclude.hlsl"
    ///      ^                          ^
    ///    Start                      End
```